math/dec_test.go

```diff
@@ -1287,3 +1287,155 @@ func must[T any](r T, err error) T {
 	}
 	return r
 }
+
+func TestMarshal(t *testing.T) {
+	specs := map[string]struct {
+		x   Dec
+		exp string
+	}{
+		"No trailing zeros": {
+			x:   NewDecFromInt64(123456),
+			exp: "123456",
+		},
+		"Trailing zeros": {
+			x:   NewDecFromInt64(123456000),
+			exp: "123456000",
+		},
+		"Zero value": {
+			x:   NewDecFromInt64(0),
+			exp: "0",
+		},
+		"Decimal value": {
+			x:   must(NewDecFromString("1.30000")),
+			exp: "1.30000",
+		},
+		"Positive value": {
+			x:   NewDecFromInt64(10),
+			exp: "10",
+		},
+		"negative value": {
+			x:   NewDecFromInt64(-10),
+			exp: "-10",
+		},
+		"max decimal": {
+			x:   must(NewDecFromString("9." + strings.Repeat("0", 34))),
+			exp: "9.0000000000000000000000000000000000",
+		},
+		"min decimal": {
+			x:   must(NewDecFromString("-1." + strings.Repeat("0", 34))),
+			exp: "-1.0000000000000000000000000000000000",
+		},
+		"1e100000": {
+			x:   NewDecWithPrec(1, 100_000),
+			exp: "1E+100000",
+		},
+		"1.1e100000": {
+			x:   NewDecWithPrec(11, 100_000),
+			exp: "1.1E+100001",
+		},
+		"1.e100000": {
+			x:   NewDecWithPrec(1, 100_000),
+			exp: "1E+100000",
+		},
+	}
+	for name, spec := range specs {
+		t.Run(name, func(t *testing.T) {
+			marshaled, err := spec.x.Marshal()
+			require.NoError(t, err)
+			assert.Equal(t, spec.exp, string(marshaled))
+		})
+	}
+}
+
+func TestUnMarshal(t *testing.T) {
+	specs := map[string]struct {
+		x      string
+		exp    string
+		expErr error
+	}{
+		"No trailing zeros": {
+			x:   "123456",
+			exp: "123456",
+		},
+		"Trailing zeros": {
+			x:   "1.23456E+8",
+			exp: "123456000",
+		},
+		"Small e": {
+			x:   "1.23456e+8",
+			exp: "123456000",
+		},
+		"Zero value": {
+			x:   "0",
+			exp: "0",
+		},
+		"Decimal value": {
+			x:   "1.3000",
+			exp: "1.3000",
+		},
+		"Positive value": {
+			x:   "1E+1",
+			exp: "10",
+		},
+		"negative value": {
+			x:   "-1E+1",
+			exp: "-10",
+		},
+		"max decimal": {
```

Comment:
no need for this or min decimal. We should test at the boundaries with max/min exponents.